tests/virt_autotest/sriov_pci_passthrough.pm

```diff
@@ -0,0 +1,279 @@
+# SUSE's openQA tests
+ #
+ # Copyright (C) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+ 
+# Summary: A test on KVM host to pass a SR-IOV NIC VF to guest via libvirt.
+# Maintainer: Julie CAO <JCao@suse.com>
+ 
+use base "virt_feature_test_base";
+use strict;
+use warnings;
+use testapi;
+use set_config_as_glue;
+use utils;
+use virt_utils;
+use xen;
+use virt_autotest::virtual_network_utils qw(save_guest_ip test_network_interface);
+ 
+sub run_test {
+  my $self = shift;
+
+#  my $hypervisor;
+#  if (check_var("HOST_HYPERVISOR", "kvm") || check_var("SYSTEM_ROLE", "kvm") || check_var("HYPERVISOR", "kvm")) {
+#    $hypervisor = "kvm";
+#  }
+#  else {
+#    $hypervisor = "xen";
+#  }
+
+  #get the SR-IOV device BDF and interface 
+  my @host_pfs;
+  @host_pfs = get_sriov_devices ();
+  $self->{test_results}->{"host"}->{"Error: not found any SR-IOV network device in the host!"}->{status} = 'FAILED' if ! $#host_pfs;
```

Comment:
It seems this kind of test_results format is a little uncommon. Generally people define test results like $self->{test_results}->{test target, eg host or guest}->{test case or test step}->{status} =xx, and append detailed error msg like {test_results}->{test target, eg host or guest}->{test case or test step}->{error}=xx, right? But your way seems like {test_results}->{test target, eg host or guest}->{failurexx}->{status} = 'FAILED'.